src/features/page_view_timing/aggregate/index.js

```diff
@@ -94,6 +94,7 @@ export class Aggregate extends AggregateBase {
 
   addTiming (name, value, attrs) {
     attrs = attrs || {}
+    addConnectionAttributes(attrs)
```

Comment:
By moving this out of the vital wrapper, any objects pulled from the buffer will get the network information at the time of calling, not the time they were observed. 

Before:
A metric flowed through the vitalmetric class --> it was stored with network info at the time of flowing through --> saved in buffer --> future subscribers would get that value, which would have the network attributes at the time of saving.  This is likely an edge case, but i'm not sure its the best way to ensure accuracy.  Thoughts there?

Now:
By moving this out of the vital wrapper, any objects pulled from the buffer will get the network information at the time of calling, not the time they were observed. 

Before:
A metric flowed through the vitalmetric class --> it is stored without any network metrics --> saved in buffer --> future subscribers would get that value, which would have no network attributes at the time of saving, instead getting the network attributes at the time of retrieval.

Reply:
One of those both-sides debates, but imo a) the vital metric class is trying to do too much if it's also tracking network conditions and b) that's more of a job for the overall pvt (aggregate) process. Up until of late, `addTiming` functions as the central point when buffered data is processed into event units--still is--and is called directly by other non-vital timing events,  so it made the most sense to tack on network info there.
Your case specifically targets a subscriber that wants that info with past vitals (or other timings in general) which doesn't exist just yet, but if and when we get to that, the solution should be generalized to apply to all timings. If that's an issue, we need to rethink how #handleVitalmetric is calling addTiming but also having a separate system/pathway of storing another version of the same data if that makes sense. E.g., maybe addTiming should return it's final processed output and if the vital metrics class needs to store something, it should store that instead.

Reply:
We can add it in vital metrics and check in `addTiming` if it exists and add it again if it doesn't (for non-vital timings) but splitting it up that way causes confusion.

Reply:
I think it would just be simplest to add it to the vital timing at the time it was recorded.  If the consumers of the metric use or dont use it is inconsequential, because at the end of the day we just want to record it when it happens and not later, even if that gap is small or an edge case -- simply to save ourselves from ourselves later if the consumption of those singletons' subscriptions ever gets deferred in some way.